app/services/hotkeys.ts

```diff
@@ -519,31 +537,65 @@ export class Hotkey implements IHotkey {
 
 
   private getAction(entityId: string): IHotkeyAction {
-    const action = { ...HOTKEY_ACTIONS[this.type].find(action => {
-      return action.name === this.actionName;
-    }) };
+    const action = getActionFromName(this.actionName);
 
     const { up, down } = action;
 
     // Fill in optional functions with some defaults
-    if (!action.isActive) action.isActive = () => false;
-    if (!action.shouldApply) action.shouldApply = () => true;
+    if (!action.isActive) {
+      action.isActive = () => false;
+    }
+    if (!action.shouldApply) {
+      action.shouldApply = () => true;
+    }
 
 
     // We defer the actions until after we've decided whether
     // or not to execute each action.
     if (up) {
       action.upHandler = () => {
-        if (!action.isActive(entityId)) defer(() => up(entityId));
+        if (!action.isActive(entityId)) {
+          defer(() => up(entityId));
+        }
       };
     }
 
     if (down) {
       action.downHandler = () => {
-        if (!action.isActive(entityId)) defer(() => down(entityId));
+        if (!action.isActive(entityId)) {
+          defer(() => down(entityId));
+        }
       };
     }
 
     return action;
   }
 }
+
+const getMigrationMapping = (actionName: string) => {
+  return {
+    MUTE: 'TOGGLE_MUTE',
+    UNMUTE: 'TOGGLE_UNMUTE',
+    // SHOW_SCENE_ITEM: 'TOGGLE_SOURCE_VISIBILITY_SHOW',
+    // HIDE_SCENE_ITEM: 'TOGGLE_SOURCE_VISIBILITY_HIDE',
+  }[actionName.split('.')[0]];
+};
+
+const getActionFromName = (actionName: string) => {
+  const action = ACTIONS[actionName] || getMigrationMapping(actionName);
+  if (!action) {
+    console.warn(`Couldn't find action or mapping for ${actionName}`);
+  }
+  return action;
+};
+
+const isSupportedHotkey = (hotkey: OBSHotkey) =>
+  hotkey.ObjectType === obs.EHotkeyObjectType.Source && getActionFromName(hotkey.HotkeyName);
+
+const isSceneItem = (hotkey: OBSHotkey) =>
+  !!hotkey.HotkeyName.match(/^(SHOW|HIDE)_SCENE_ITEM.(.+)$/);
+
+const isSource = (hotkey: OBSHotkey) => !!hotkey.ObjectName.match(/(source|input|output)/);
+
+const idPropFor = (hotkey: OBSHotkey) =>
+  isSource(hotkey) ? 'sourceId' : isSceneItem(hotkey) ? 'sceneItemId' : 'sceneId';
```

Comment:
IMO nested ternaries make the logic difficult to follow.  We don't have a strict style guide (yet) but started out loosely following AirBnB, which recommends against this: https://github.com/airbnb/javascript#comparison--nested-ternaries

Reply:
Ideally we would not need this at all. I think ternaries are a common `if` replacement (which we should hate more 😂) but I'm happy to remove them.